client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -353,13 +366,17 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 			selectedDesignThemeId &&
 			getIsMarketplaceThemeSubscribed( state, selectedDesignThemeId, site.ID )
 	);
+	const isExternallyManagedThemeAvailable = useSelector(
+		( state ) => site?.ID && isSiteEligibleForManagedExternalThemes( state, site.ID )
+	);
 
 	const isPluginBundleEligible = useIsPluginBundleEligible();
 	const isBundledWithWooCommerce = selectedDesign?.is_bundled_with_woo_commerce;
 
 	const shouldUpgrade =
```

Comment:
Nitpick (non-blocking): Previously, this variable only was true if the user didn't have the needed plan and thus should upgrade. But that has changed now: a user on a Business won't need to upgrade but purchase the theme, so we might want to rename this to `isLockedTheme` or something like that to prevent incorrect assumptions from future developers reading this code.

Reply:
Done on https://github.com/Automattic/wp-calypso/pull/80988/commits/891a4fecf766016a7cb03baf50ab5e4f6bb3b040